pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -238,6 +245,79 @@ def mul_in_parallel(first: Resources, scalar: int, in_place=False) -> Resources:
     return Resources(new_wires, new_gates, new_gate_types)
 
 
+def substitute(
+    initial_resources: Resources, name: str, replacement_resources: Resources, in_place=False
+) -> Resources:
+    """Replaces a specified gate in a :class:`~.resource.Resources` object with the contents of another :class:`~.resource.Resources` object.
+
+    Args:
+        initial_resources (Resources): the resources to be modified
+        name (str): the name of the operation to be replaced
```

Comment:
Please make sure that the arg name is consistent with what you have elsewhere. Maybe gate_name instead of name?